server-base/src/main/java/org/eclipse/hono/serverbase/HonoServerBase.java

```diff
@@ -0,0 +1,161 @@
+package org.eclipse.hono.serverbase;
+
+/**
+ * Copyright (c) 2017 Bosch Software Innovations GmbH.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Bosch Software Innovations GmbH - initial creation
+ */
+
+import io.vertx.core.AbstractVerticle;
+import io.vertx.core.Future;
+import org.eclipse.hono.config.HonoConfigProperties;
+import org.eclipse.hono.util.Constants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Objects;
+
+public abstract class HonoServerBase extends AbstractVerticle {
+
+    private static final Logger LOG = LoggerFactory.getLogger(HonoServerBase.class);
+
+    protected HonoConfigProperties config = new HonoConfigProperties();
+
+    protected int port         = Constants.PORT_UNCONFIGURED;
+    protected int insecurePort = Constants.PORT_UNCONFIGURED;
+
+    protected abstract int getPortDefaultValue();
+    protected abstract int getInsecurePortDefaultValue();
+
+    public int getPort() {
+        return port;
+    }
+
+    /*
+     * Define the default value for the TLS port in the implementing server class (e.g. 5671 for AMQP 1.0).
+    */
+
+    public int getInsecurePort() {
+        return insecurePort;
+    }
+
+    /*
+     * Define the default value for the not encrypted port in the implementing server class (e.g. 5672 for AMQP 1.0).
+     */
+
+
+    public void setConfig(final HonoConfigProperties props) {
+        this.config = Objects.requireNonNull(props);
+    }
+
+
+    /**
+     * Either one secure, one insecure or one secure AND insecure ports are supported for servers in Hono.
+     *
+     * @param portConfigurationTracker The future to continue or fail (in case of configuration errors).
+     * @throws NullPointerException if portConfigurationTracker is null.
+     */
+    public void determinePortConfigurations(Future<Void> portConfigurationTracker) {
```

Comment:
Actually, it is only public due to access for tests from a different package.

I will remove the tests (that are not essentially necessary) there and reduce the method visibility to what it should be: protected.